docs/design/2020-04-20-brie.md

```diff
@@ -0,0 +1,331 @@
+
+Because the BRIE statements interact with TiKV directly and can read from / write to arbitrary path,
+all 4 statements require SUPER privilege to execute.
+
+If the statement is canceled or an error is encountered, it will report an error e.g.
+```
+MySQL [tpcc]> backup table tpcc.stock to 'local:///tmp/br_tpcc_30';
+ERROR 8124 (HY000): Backup failed: backup meta exists, may be some backup files in the path already
+```
+
+If the statement completed successfully, it will report one row of summary e.g.
+```
+MySQL [tpcc]> backup database tpcc to 'local:///tmp/br_tpcc_32';
++-------------------------+-----------+-----------------+---------------------+---------------------+
+| Destination             | Size      | BackupTS        | Queue_Time          | Execution_Time      |
++-------------------------+-----------+-----------------+---------------------+---------------------+
+| local:///tmp/br_tpcc_32 | 248665063 | 416099531454472 | 2020-04-12 23:09:48 | 2020-04-12 23:09:48 |
++-------------------------+-----------+-----------------+---------------------+---------------------+
+1 row in set (58.453 sec)
+```
+
+> **Note:**
+>
+> The `RESTORE` (BR) and `IMPORT` (Lightning) operations are not using transactions and are not ACID compliant.
+>
+> The data and index KV pairs are ingested separately, i.e. a table being restored would have in an intermediate state
+> where some rows are missing index, and some index doesn't have corresponding rows.
+> If the operation has failed or canceled, the table would be stuck in this inconsistent state.
+>
+> There is currently no plan to change this behavior.
+
+### Status query
+
+The status of the queued and running tasks can be queried using the `SHOW` statement:
+
+```sql
+SHOW BACKUPS;
+SHOW RESTORES;
+SHOW IMPORTS;
+SHOW EXPORTS;
+```
+
+Example:
+```
+MySQL [(none)]> show backups;
++-------------------------+---------+-------------------+---------------------+---------------------+------------+
+| Destination             | State   | Progress          | Queue_Time          | Execution_Time      | Connection |
++-------------------------+---------+-------------------+---------------------+---------------------+------------+
+| local:///tmp/br_tpcc_30 | Backup  | 98.38709677419355 | 2020-04-12 23:09:03 | 2020-04-12 23:09:25 |          4 |
+| s3://bucket/prefix      | Wait    |                 0 | 2020-04-12 23:09:48 | 2020-04-12 23:09:48 |          9 |
++-------------------------+---------+-------------------+---------------------+---------------------+------------+
+```
+
+The content can be filtered using `LIKE` and `WHERE` clauses, e.g.
+```sql
+SHOW BACKUP LIKE 'local:%';  -- same as `Destination` LIKE 'local:%'
+SHOW BACKUP WHERE `State` = 'Wait';
+```
+
+Similar to the BRIE statements, these `SHOW` statements also require SUPER privilege.
+
+To protect against leakage of secret keys,
+all query parameters of S3 and GCS destinations are discarded in the SHOW statements.
```

Comment:
`INFORMATION_SCHEMA.processlist` and `show processlist` should also be considered.

Reply:
Does `INFORMATION_SCHEMA.processlist` use [`Restore()`](https://godoc.org/github.com/pingcap/parser/ast#Node)? We could redact the URL in `Restore()` before GA.

Reply:
I think it's too late.
After an executor is Open, TiDB sets the original SQL into `INFORMATION_SCHEMA.processlist`. 
https://github.com/pingcap/tidb/blob/7106b189df620bbaf264509002f8a3939956198a/executor/adapter.go#L341-L349

`ast.SensitiveStmtNode` can be considered to use.

Reply:
@SunRunAway filed pingcap/parser#842 for that. but these two PRs don't block each other.